.github/settings.yml

```diff
@@ -0,0 +1,119 @@
+  #has_issues: true
+
+  # Either `true` to enable projects for this repository, or `false` to disable them.
+  # If projects are disabled for the organization, passing `true` will cause an API error.
+  #has_projects: true
+
+  # Either `true` to enable the wiki for this repository, `false` to disable it.
+  #has_wiki: false
+
+  # Either `true` to enable downloads for this repository, `false` to disable them.
+  #has_downloads: true
+
+  # Updates the default branch for this repository.
+  #default_branch: main
+
+  # Either `true` to allow squash-merging pull requests, or `false` to prevent
+  # squash-merging.
+  #allow_squash_merge: true
+
+  # Either `true` to allow merging pull requests with a merge commit, or `false`
+  # to prevent merging pull requests with merge commits.
+  #allow_merge_commit: true
+
+  # Either `true` to allow rebase-merging pull requests, or `false` to prevent
+  # rebase-merging.
+  #allow_rebase_merge: true
+
+  # Either `true` to enable automatic deletion of branches on merge, or `false` to disable
+  #delete_branch_on_merge: false
+
+  # Either `true` to enable automated security fixes, or `false` to disable
+  # automated security fixes.
+  #enable_automated_security_fixes: true
+
+  # Either `true` to enable vulnerability alerts, or `false` to disable
+  # vulnerability alerts.
+  #enable_vulnerability_alerts: true
+
+# Collaborators: give specific users access to this repository.
+# See https://docs.github.com/en/rest/reference/repos#add-a-repository-collaborator for available options
+#collaborators:
+  # - username: bkeepers
+  #   permission: push
+  # - username: hubot
+  #   permission: pull
+
+  # Note: `permission` is only valid on organization-owned repositories.
+  # The permission to grant the collaborator. Can be one of:
+  # * `pull` - can pull, but not push to or administer this repository.
+  # * `push` - can pull and push, but not administer this repository.
+  # * `admin` - can pull, push and administer this repository.
+  # * `maintain` - Recommended for project managers who need to manage the repository without access to sensitive or destructive actions.
+  # * `triage` - Recommended for contributors who need to proactively manage issues and pull requests without write access.
+
+# See https://docs.github.com/en/rest/reference/teams#add-or-update-team-repository-permissions for available options
+# teams:
+    # The permission to grant the team. Can be one of:
+    # * `pull` - can pull, but not push to or administer this repository.
+    # * `push` - can pull and push, but not administer this repository.
+    # * `admin` - can pull, push and administer this repository.
+    # * `maintain` - Recommended for project managers who need to manage the repository without access to sensitive or destructive actions.
+    # * `triage` - Recommended for contributors who need to proactively manage issues and pull requests without write access.
+  # - name: org-admins
+  #  permission: admin
```

Comment:
Do we really need both `homebrew-admins` and `homebrew-tap-admins` here? 

Do we need `homebrew-tap-contributors` and `fsoc-admins` granted write access here?

Reply:
I removed this section because last time I tried it permission `write` did not work (maybe `push` is the right terminology)

> Do we really need both homebrew-admins and homebrew-tap-admins here?

Permissions for `homebrew-admins` can be removed, this team is deprecated

> Do we need homebrew-tap-contributors and fsoc-admins granted write access here?

`fsoc-admins` are code owners of their formulae, to be code owners they need write permissions otherwise github is complaining about missing permissions. What I *actually* want to implement is that they get asked to review changes to their formulae & approve them but not merge them

Reply:
> `fsoc-admins` are code owners of their formulae, to be code owners they need write permissions otherwise github is complaining about missing permissions. What I actually want to implement is that they get asked to review changes to their formulae & approve them but not merge them.

Any particular reason why you would want to restrict them from merging them? If the person who submitted the PR didn't want it merged after review, they could just create a draft PR...  I also don't see the need for an additional third party gatekeeper to merge, since we could introduce required automated checks to block merging for scenarios we'd like to prevent (malformed config, etc)

Reply:
> Any particular reason why you would want to restrict them from merging them? If the person who submitted the PR didn't want it merged after review, they could just create a draft PR... I also don't see the need for an additional third party gatekeeper to merge, since we could introduce required automated checks to block merging for scenarios we'd like to prevent (malformed config, etc)

The PR is raised via @cisco-service, so the person approving is by proxy the same person triggering that PR (via go-releaser). It _feels_ like there should be an extra layer of review, but yes, probably we are fine

Reply:
Ah, gotcha. I still don't see it as an issue, since there would be proper review and approval for the fsoc release prior to the push event that is triggering the GitHub action creating this PR.

That said, I think there is currently a loophole in that we aren't actually blocking folks from pushing tags, as long as they have the appropriate permissions to do so.

It seems like longer term, we'd either want to:
- Update homebrew updates as a workflow on this repo, watching for new releases on remote repos
- Use rulesets to limit tag pushes (https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-rulesets/creating-rulesets-for-a-repository#selecting-branch-or-tag-protections) and move the creation of the tag to an automated process that requires a separate reviewer.

Reply:
As an aside, every time I've tried to add a full settings.yml configuration to a repo, I've had issues with the settings bot throwing errors (that are not exposed) and failing to apply the full config. You may need to apply the config incrementally (https://github.com/repository-settings/app/issues/518)

Reply:
> As an aside, every time I've tried to add a full settings.yml configuration to a repo, I've had issues with the settings bot throwing errors (that are not exposed) and failing to apply the full config. You may need to apply the config incrementally ([repository-settings/app#518](https://github.com/repository-settings/app/issues/518))

Oh... good to know, then let me maybe split this up into multiple PRs, makes the discussions here easier as well :)